### PR TITLE
webclient: implement basic proxy

### PR DESCRIPTION
**Pull request by yamt**
## Summary
webclient: implement basic proxy

## Testing
a ported version tested on macOS

### CODE REVIEW

File: netutils/webclient/webclient.c [hunk 1]
```diff
@@ -182,15 +182,21 @@ struct conn_s
 #define	WGET_FLAG_GOT_CONTENT_LENGTH 1U
 #define	WGET_FLAG_CHUNKED            2U
 
+struct wget_target_s
+  {
+    char scheme[sizeof("https") + 1];
+    char hostname[CONFIG_WEBCLIENT_MAXHOSTNAME];
+    char filename[CONFIG_WEBCLIENT_MAXFILENAME];
+    uint16_t port;     /* The port number to use in the connection */
+  };
```
Comment (pkarashchenko) [inline netutils/webclient/webclient.c:191]:
```suggestion
{
  char scheme[sizeof("https") + 1];
  char hostname[CONFIG_WEBCLIENT_MAXHOSTNAME];
  char filename[CONFIG_WEBCLIENT_MAXFILENAME];
  uint16_t port;     /* The port number to use in the connection */
};
```
BTW, do you know why `scheme` is `sizeof("https") + 1` and not `sizeof("https")`?

File: netutils/webclient/webclient.c [hunk 2]
```diff
@@ -1300,9 +1358,21 @@ int webclient_perform(FAR struct webclient_context *ctx)
                 {
                   /* Get the server address from the host name */
 
+                  struct wget_target_s *target;
```
Comment (pkarashchenko) [inline netutils/webclient/webclient.c:1361]:
```suggestion
                  FAR struct wget_target_s *target;
```

Comment (yamt) [reply netutils/webclient/webclient.c]: done

**Review by pkarashchenko — APPROVED** [review]

File: netutils/webclient/webclient.c (hunk 1 above)
Comment (pkarashchenko) [reply netutils/webclient/webclient.c]: seems like +1 is not needed. I will add checking this to my TODO list :)

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**